GoogleForCreators/web-stories-wp
# Library: only render tabs when active

## Context
- To enhance performance, do not execute the render of media, template, and text tabs.
- Can be tested by looking at the network tab. On main, the editor will load text set and template jsons before those tabs are selected. In this PR the editor does not load those until the tabs are selected.


## Summary
- add a reference to track which tabs have been rendered to determine when to render their content
- once the pane is rendered the first time, leave it rendered
- change the get text sets effect to check if the text pane has been rendered


## Relevant Technical Choices
- I only load the text sets once to reduce the network load. 
- Due to the high render time of the text sets, per Morten's comment on the ticket, we switch to manual activation for the tabs instead of selection follows focus if switching "over" the text tab (using keyboard) is too slow.
- However, from manual testing this seems to be faster



## To-do


## User-facing changes
N/A


## Testing Instructions



### QA
This is a performance enhancement.




1. Open a new story and observe the network tab in your inspector tools
2. Click on each of the library tabs
3. The templates and media should load (seen in network tab) after clicking on their respective library tabs
4. The templates and media should not re-load when clicking on the tabs a second time before refreshing
### UAT








## Reviews

### Does this PR have a security-related impact?
No


### Does this PR change what data or activity we track or use?
No


### Does this PR have a legal-related impact?
No


## Checklist




---



Fixes #4634
